src/main/java/com/GeorgesServer/app/MyServer.java

```diff
@@ -1,31 +1,34 @@
 package com.GeorgesServer.app;
 
 import com.GeorgesServer.app.com.GeorgesServer.handler.IHandler;
+import com.GeorgesServer.app.com.GeorgesServer.request.ClientRequest;
+import com.GeorgesServer.app.com.GeorgesServer.response.ServerResponse;
 
 public class MyServer {
     private StreamMaker streamMaker;
     private RequestParser requestParser;
-    private IHandler requestHandler;
     private ResponseSender responseSender;
+    private Router router;
 
     public MyServer(StreamMaker streamMaker,
                     RequestParser requestParser,
-                    IHandler requestHandler,
                     ResponseSender responseSender,
+                    Router router,
                     String publicFolderPath) {
 
         this.streamMaker = streamMaker;
         this.requestParser = requestParser;
-        this.requestHandler = requestHandler;
         this.responseSender = responseSender;
+        this.router = router;
     }
 
     public void start() {
         String formattedResponse = "";
         while (!formattedResponse.equals("Bye")) {
             Streams streams = streamMaker.connect();
             ClientRequest clientRequest = requestParser.parse(streams.getIn());
-            ServerResponse serverResponse = requestHandler.handle(clientRequest);
+            IHandler handler = router.route(clientRequest.getMethod(), clientRequest.getUrl());
```

Comment:
Is there a reason you aren't passing in the whole request?

Reply:
actually no...I'll correct that in my next PR.